### PR TITLE
Git_Help.rst: Add remote repo sync section

### PR DESCRIPTION
**Pull request by arafsheikh**
Fixes https://github.com/coala-analyzer/coala/issues/1774

### CODE REVIEW

**Comment by arafsheikh** [comment]
@Makman2 do you mean I should add a link to those guides?

**Comment by Makman2** [comment]
yes

**Comment by arafsheikh** [comment]
Didn't add second link as that part is already explained in the guide.

**Comment by SanketDG** [comment]
@arafsheikh A great idea would be to record an asciinema, and link to it(no embed).

File: docs/Users/Tutorials/Git_Help.rst [hunk 1]
```diff
@@ -230,6 +220,27 @@ fine by our reviewers, you can simply send it again like this. If not, edit it
...
+-------------------------
+
+As people work on coala new commits will be added. This will result in your
+local fork going out of sync with the remote repository.
+To sync your changes with the remote repository run the following commands:
+
+::
+
+    $ git checkout master
+    $ git git pull origin master
```
Comment (SanketDG) [inline docs/Users/Tutorials/Git_Help.rst]: `git` repeated. Although I did advise against using `git pull`, read http://stackoverflow.com/questions/292357/what-are-the-differences-between-git-pull-and-git-fetch and http://longair.net/blog/2009/04/16/git-fetch-and-merge/

The correct way to keep a sync:

```shell
$ git fetch upstream
$ git checkout branch
$ git rebase upstream/master
```

Comment (SanketDG) [inline docs/Users/Tutorials/Git_Help.rst]: Also, please explain this whole process. A newcomer will not probably have a remote named `upstream`.

Comment (AbdealiJK) [inline docs/Users/Tutorials/Git_Help.rst]: Oh ... I normally use the 4 commands mentioned in the docs so suggested those.
Anything is fine as long as it works

File: docs/Users/Tutorials/Git_Help.rst [hunk 2]
```diff
@@ -230,6 +220,27 @@ fine by our reviewers, you can simply send it again like this. If not, edit it
...
+    $ git checkout branchname
+    $ git rebase master
+
+Explaination on what each command does can be found below.
+
+.. seealso::
+
+  For more information about syncing a fork, see
+  `Github's guide <https://help.github.com/articles/syncing-a-fork/>`_.
```
Comment (SanketDG) [inline docs/Users/Tutorials/Git_Help.rst]: Not needed. Your workflow is different, you might want to link to a couple of SO questions.

File: docs/Users/Tutorials/Git_Help.rst [hunk 3]
```diff
@@ -281,6 +292,11 @@ files are untracked with git status and remove them manually. Like an ordinary
 rm command, ``git clean`` is not undoable, so make sure you really want to
 delete the untracked files before you run it.
 
+::
+
+    $ git checkout
```
Comment (SanketDG) [inline docs/Users/Tutorials/Git_Help.rst]: Specify the required branch, and mention it in the description.
```
$ git checkout <branch>
```

File: docs/Users/Tutorials/Git_Help.rst (hunk 1 above)
Comment (AbdealiJK) [inline docs/Users/Tutorials/Git_Help.rst]: @SanketDG your upstream is the origin in this tutorial. So,
can they just do
```
git fetch origin
git rebase origin/master 
```
While they are in the `branchname` branch ?

Comment (SanketDG) [inline docs/Users/Tutorials/Git_Help.rst]: @AbdealiJK yes, this is the safest way.

Comment (arafsheikh) [inline docs/Users/Tutorials/Git_Help.rst]: This is what I originally suggested. I'll revert back to this.

Comment (AbdealiJK) [inline docs/Users/Tutorials/Git_Help.rst]: @arafsheikh Sorry - Seems like my git knowledge isn't very good

File: docs/Users/Tutorials/Git_Help.rst [hunk 4]
```diff
@@ -230,6 +220,20 @@ fine by our reviewers, you can simply send it again like this. If not, edit it
...
+As people work on coala new commits will be added. This will result in your
+local fork going out of sync with the remote repository.
+To sync your changes with the remote repository run the following commands:
+
+::
+
+    $ git fetch origin
+    $ git rebase origin/master
+
+Explaination on what each command does can be found below.
```
Comment (SanketDG) [inline docs/Users/Tutorials/Git_Help.rst]: Explanation

File: docs/Users/Tutorials/Git_Help.rst [hunk 5]
```diff
@@ -281,6 +285,12 @@ files are untracked with git status and remove them manually. Like an ordinary
 rm command, ``git clean`` is not undoable, so make sure you really want to
 delete the untracked files before you run it.
 
+::
+
+    $ git checkout <branch>
+
+The ``git checkout`` command is used to switch to another branch in the
+repository. Here <branch> is the name of the branch to switch to.
```
Comment (SanketDG) [inline docs/Users/Tutorials/Git_Help.rst]: Here <branch> is the name of the branch you want to switch to

File: docs/Users/Tutorials/Git_Help.rst [hunk 6]
```diff
@@ -230,6 +220,20 @@ fine by our reviewers, you can simply send it again like this. If not, edit it
 and send it.
 Now you have successfully edited your last commit!
 
+Keeping your fork in sync
+-------------------------
+
+As people work on coala new commits will be added. This will result in your
+local fork going out of sync with the remote repository.
+To sync your changes with the remote repository run the following commands:
```
Comment (SanketDG) [inline docs/Users/Tutorials/Git_Help.rst]: run the following commands in the desired branch

**Comment by sils1297** [comment]
also rebase please

**Comment by sils1297** [comment]
ack 545878e

**Comment by sils1297** [comment]
@rultor merge

**Pull request merged**